x-pack/plugins/watcher/common/types/watch_types.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export interface ExecutedWatchResults {
+  id: string;
+  watchId: string;
+  details: any;
+  startTime: Date;
+  watchStatus: {
+    state: string;
+    actionStatuses: [
+      {
+        state: string;
+        lastExecutionReason: string;
+      }
+    ];
+  };
+}
+
+export interface ExecutedWatchDetails {
+  triggerData: {
+    triggeredTime: Date;
+    scheduledTime: Date;
+  };
+  ignoreCondition: boolean;
+  alternativeInput: any;
+  actionModes: {
+    [key: string]: string;
+  };
+  recordExecution: boolean;
+  upstreamJson: any;
+}
+
+export interface BaseWatch {
```

Comment:
we should probably just move all the models to TS, but we can do that in another PR.

Reply:
i can work on that next